internal/logger/target/http/http.go

```diff
@@ -374,30 +368,30 @@ func (h *Target) startQueueProcessor(ctx context.Context, mainWorker bool) {
 					h.failedMessages.Add(1)
 					continue
 				}
-				count++
 			} else {
 				entries = append(entries, entry)
-				count++
 			}
-		}
+			count++
 
-		if count != h.batchSize {
-			if len(h.logCh) > 0 || len(globalBuffer) > 0 || count == 0 {
+			if len(h.logCh) > 0 || len(globalBuffer) > 0 {
 				continue
 			}
 
-			if h.batchSize > 1 {
-				// If we are doing batching, we should wait
-				// at least one second before sending.
-				// Even if there is nothing in the queue.
-				if time.Since(lastBatchProcess).Seconds() < 1 {
-					continue
-				}
+			// If we are doing batching, we should wait
+			// at least for 2 seconds before sending.
+			// Even if there is nothing in the queue.
+			if h.batchSize > 1 && time.Since(lastBatchProcess).Seconds() < 2 {
```

Comment:
we probably need to keep the ticker here, because we may wait for a long time until next event comes in

Reply:
no this only waits since the lastBatchProcess - we sent at once whatever the count is.